msprime/mutations.py

```diff
@@ -199,6 +199,285 @@ def __init__(self, state_independent=False):
+    transition matrix, we define :math:`M = \\max_i \\left( \\sum_j q_{ij} \\right)`.
+    Then, as a :class:`.MatrixMutationModel` this model has the transition
+    matrix :math:`P_{ij} = q_{ij} / M`, and :math:`P_{ii} = 1 - \\sum_{j\\neq i} P_{ij}`.
+
+    :param float s: strength of length dependence on mutation rate
+     (i.e. s=0 uniform rate)
+    :param float u: constant bias parameter; U ~ (0, 1)
+    :param float v: linear bias parameter; v ~ (-inf, inf)
+    :param float p: probability of a single step mutation
+    :param float m: success prob of truncated gamma distribution
+    :param int lo: lower bound on repeat number
+    :param int hi: upper bound on repeat number
+
+    :return: rate matrix
+
+    """
+
+    nalleles = hi - lo + 1
+    rate_matrix = np.zeros((nalleles, nalleles))
+    for i in range(lo, hi + 1):
+        for j in range(lo, hi + 1):
+            # shift coords to account for 0-indexing in rate matrix
+            ii = i - lo
+            jj = j - lo
+            if i == j - 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * alpha_microsat(i, u, v, lo)
+                    * (p + ((1 - p) * gamma_microsat(m, i, j, lo, hi)))
+                )
+            elif i < j - 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * alpha_microsat(i, u, v, lo)
+                    * (1 - p)
+                    * gamma_microsat(m, i, j, lo, hi)
+                )
+            elif i == j + 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * (1 - alpha_microsat(i, u, v, lo))
+                    * (p + ((1 - p) * gamma_microsat(m, i, j, lo, hi)))
+                )
+            elif i > j + 1:
+                rate_matrix[ii, jj] = (
+                    beta_microsat(i, s, lo)
+                    * (1 - alpha_microsat(i, u, v, lo))
+                    * (1 - p)
+                    * gamma_microsat(m, i, j, lo, hi)
+                )
+            else:
+                rate_matrix[ii, jj] = 0
+    # scale by max row sum; then normalize to 1
+    row_sums = rate_matrix.sum(axis=1, dtype="float64")
+    alpha = max(row_sums)
+    rate_matrix /= alpha
+    row_sums = rate_matrix.sum(axis=1, dtype="float64")
+    np.fill_diagonal(rate_matrix, 1 - row_sums)
+    return rate_matrix
+
+
+class SMM_MicrosatMutationModel(MatrixMutationModel):
+    """
+    Stepwise mutation model (Kimura and Ohta, 1978),
+    for microsatellite repeats number.
+
+    This is a :class:`.MatrixMutationModel` with alleles ``[lo, .. , hi]``,
+    root distribution is uniform ``(lo, hi)``, and a transition matrix that reflects
+    single-step mutation.
+
+    Note that the values of ``lo`` and ``hi`` are in units of repeat number,
+    and so can be arbitrarily large with respect to repeat length
+    (e.g. dinucleotide, tri-nucleotide, etc.).
```

Comment:
```suggestion
    Note that the values of ``lo`` and ``hi`` are in units of repeat number,
    and are unrelated to repeat length (e.g., the repeat itself can be dinucleotide, tri-nucleotide, etcetera).
```
Is this what you meant?